astroplan/core.py

```diff
@@ -25,82 +40,137 @@ class Observer(object):
+        pressure : `~astropy.units.Quantity`
+            The ambient pressure.
 
-        latitude : str or `~astropy.units.quantity`?
-            The latitude of the observing location.
+        location : `~astropy.coordinates.EarthLocation`
+            The location (latitude, longitude, elevation) of the observatory.
+
+        longitude : str or `~astropy.units.Quantity`
+            The longitude of the observing location. If str, should be a string
+            that initializes a `~astropy.coordinates.Longitude` object with
+            units in degrees.
+
+        latitude : str or `~astropy.units.Quantity`
+            The latitude of the observing location. If str, should be a string
+            that initializes a `~astropy.coordinates.Latitude` object with
+            units in degrees.
 
         elevation : `~astropy.units.Quantity`
             The elevation of the observing location, with respect to sea
             level.
 
-        pressure : `~astropy.units.Quantity`
-            The ambient pressure.
-
         relative_humidity : float
             The ambient relative humidity.
 
         temperature : `~astropy.units.Quantity`
             The ambient temperature.
 
-        timezone : WHAT TYPE IS pytz.timezone ?
-            The local timezone.
+        timezone : str or `datetime.tzinfo`
+            The local timezone, as either an instance of `pytz.timezone()` or
+            the string accepted by `pytz.timezone()`.
 
         description : str
             A short description of the telescope, observatory or observing
             location.
         """
-        raise NotImplementedError()
 
-    def set_environment(self, pressure, relative_humidity, temperature):
-        """
-        Updates the Observer object with environmental conditions.
+        if name is not None:
+            self.name = name
 
-        <longer description>
+        if pressure is None:
+            pressure = 0
 
-        Parameters
-        ----------
-        pressure : `~astropy.units.Quantity`
-            The ambient pressure.
+        if temperature is None:
+            temperature = 0 * u.deg_C
 
-        relative_humidity : `~astropy.units.Quantity`
-            The ambient relative humidity.
+        if relative_humidity is None:
+            relative_humidity = 0
 
-        temperature :
-            The ambient temperature.
-        """
-        raise NotImplementedError()
+        self.pressure = pressure
+        self.temperature = temperature
+        self.relative_humidity = relative_humidity
 
-    def get_date(self, input_date_time, timezone=None):
-        """
-        Builds an object containing date and time information.
+        # If lat/long given instead of EarthLocation, convert them
+        # to EarthLocation
+        if location is None and (latitude is not None and longitude is not None):
+            accepted_latlon_types = [str, Quantity]
+            if (type(latitude) in accepted_latlon_types and
```

Comment:
Actually, it's useful for developers to know, but I think it is not in the docs because we want to encourage users to use quantities or objects (rather than strings! strings are baaaad! bad!)